### PR TITLE
Add DCO and CLA ignore note to the contributor guideline

### PR DESCRIPTION
**Pull request by flaviuvadan**
See conversation in CNCF Slack workspace: https://cloud-native.slack.com/archives/C020XM04CUW/p1666353572915679

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -5,6 +5,10 @@ appreciates your willingness to dedicate some time to Hera for the benefit of ev
 
 Please keep in mind the following guidelines and practices when contributing to Hera:
 
+1. Your commit must be signed. Hera uses [an application](https://github.com/apps/dco) that enforces the Developer 
+   Certificate of Origin (DCO). Currently, a Contributor License Agreement 
+   ([CLA](https://github.com/cla-assistant/cla-assistant)) check also appears on submitted pull requests. This can be
+   safely ignored as the Argo Project is slowly migrating projects to DCO
```
Comment (samj1912) [inline CONTRIBUTING.md:11]:
```suggestion
   safely ignored and is **not** a requirement for contribution to hera-workflows. This is an artifact as the Argo Project is slowly migrating projects from CLA to DCO. 
```

**Review by samj1912 — COMMENTED** [review]
Thanks! Could we also update the Code of conduct link to point to the one used by Argo (CNCF coc) and update the typo of `horovod` to `hera-workflows` in the last line?

**Comment by flaviuvadan** [comment]
@samj1912 updated COC and guidelines in https://github.com/argoproj-labs/hera-workflows/pull/321/commits/f2b1497ae13b3d6bce0c4e8082c71d79cef1d4e6

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -19,4 +23,4 @@ Please keep in mind the following guidelines and practices when contributing to
 ### Code of Conduct
 
 Please be mindful of and adhere to the Linux Foundation's
-[Code of Conduct](https://lfprojects.org/policies/code-of-conduct) when contributing to Horovod.
+[Code of Conduct](https://lfprojects.org/policies/code-of-conduct) when contributing to hera-workflows.
```
Comment (samj1912) [inline CONTRIBUTING.md:26]:
```suggestion
[Code of Conduct](https://github.com/cncf/foundation/blob/main/code-of-conduct.md) when contributing to hera-workflows.
```

Should this be CNCF as well?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -19,4 +23,4 @@ Please keep in mind the following guidelines and practices when contributing to
 ### Code of Conduct
 
 Please be mindful of and adhere to the Linux Foundation's
```
Comment (samj1912) [inline CONTRIBUTING.md:25]:
```suggestion
Please be mindful of and adhere to the CNCF's
```

**Review by samj1912 — APPROVED** [review]

**Pull request merged**